File: dev-docs/snippets/me/upload-avatar.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Upload an avatar image and get back the new avatar URL. `contentType` is a
// typed union of the allowed MIME types.
export async function uploadAvatar(client: JsBaoClient, image: Blob) {
  // #region example
  const { avatarUrl } = await client.me.uploadAvatar(image, "image/png");
  // #endregion example
  return avatarUrl;
}

File: dev-docs/snippets/me/upload-avatar.swift
import Foundation
import JsBaoClient

// Upload an avatar image and read back the new URL. Swift takes raw `Data` plus
// a typed `AvatarContentType` enum (`.png` / `.jpeg` / `.gif` / `.webp`) and
// returns a typed `AvatarUploadResult`.
func uploadAvatar(client: JsBaoClient, image: Data) async throws {
  // #region example
  let result = try await client.me.uploadAvatar(imageData: image, contentType: .png)
  let avatarUrl = result.avatarUrl
  // #endregion example
  _ = avatarUrl
}
